### PR TITLE
[WIP] Add support for service component

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

Add support for service component and patch `hash all` command to support service hashing

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/cli/ipfs_hash.py [hunk 1]
```diff
@@ -269,7 +279,13 @@ def check_same_ipfs_hash(
     return result
 
 
-def check_hashes(packages_dir: Path, no_wrap: bool = False,) -> int:
+def check_hashes(
+    packages_dir: Path,
+    no_wrap: bool = False,
+    config_loader: Callable[
+        [PackageType, Path], PackageConfiguration
+    ] = load_configuration,
```
Comment (angrybayblade) [inline aea/cli/ipfs_hash.py:287]: To patch the config loader on swarm

File: aea/configurations/loader.py [hunk 2]
```diff
@@ -290,16 +318,42 @@ def _load_agent_config(self, file_pointer: TextIO) -> AgentConfig:
         configuration_file_jsons = yaml_load_all(file_pointer)
         return self.load_agent_config_from_json(configuration_file_jsons)
 
+    def _load_service_config(self, file_pointer: TextIO) -> PackageConfiguration:
+        """Load an agent configuration."""
```
Comment (DavidMinarsch) [inline aea/configurations/loader.py:322]:
```suggestion
        """Load a service configuration."""
```

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]
Please add tests for the new functionality.

**Review by 8ball030 — APPROVED** [review]
LGTM

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**